src/common/snippets/include/snippets/lowered/expression_factory.hpp

```diff
@@ -49,6 +53,15 @@ class LinearIR::ExpressionFactory {
     static ExpressionPtr create(const std::shared_ptr<op::LoopEnd>& n, const std::vector<PortConnectorPtr>& inputs, const LinearIR& linear_ir);
     static ExpressionPtr create(const std::shared_ptr<ov::Node>& n, const std::vector<PortConnectorPtr>& inputs, const LinearIR& linear_ir);
 
+    // Note: PerfCountBegin nodes have a PerfCountEnd ov::Output, but corresponding expression should not have any outputs to avoid register allocation
```

Comment:
to avoid register allocation, what do you think if we extend opRegType to have like "none2none", which means no need allocate any registers, for case no emit_code is need(such as reshape/shapeof snippets node)  or emit_code() don't need register.

Reply:
In general, all connectors should be assigned a register, since the connector represent data dependency.
There are a few exceptions from this rule (and Reshape is one of them, RankNormalization - is another example), but these are mostly shape-propagation-related operations. So for now I think it would be more consistent to avoid register allocation on per-operation (expression) granularity, not on the per-connector basis.
Anyway, thank you for the suggestion! We might implement it this way, but only if we really need this flexibility.